plugins/osgi/src/main/java/org/apache/struts2/osgi/OsgiUtil.java

```diff
@@ -92,4 +92,83 @@ public static boolean containsBean(Object beanFactory, String beanId) {
 
         return false;
     }
+

```

Comment:
What about using ready to use java version resolvers like apache commons or etc via googling?

Reply:
Hi.  Thanks for the comments above, it helped point out an issue with the cache directory creation logic.  :)

In terms of using pre-existing Java version resolvers, I decided to "stay custom" within the plugin.  The main reasoning behind that was:
  1) To limit additional dependencies for the plugin at runtime (sometimes makes it more difficult to figure out why the OSGi container cannot start or load bundles), and 
  2) During testing I found that we need control over how the version was discovered, in order to be consistent with the OSGi container (with external java version resolvers, they may return a different version value than the one we need for the lookups to succeed).

So, for now at least, it seems safer to stick with the custom local Java version resolvers.

Reply:
:+1: